server/src/main/java/com/paypal/selion/grid/FileExtractor.java

```diff
@@ -174,4 +176,68 @@ static String getFileNameFromPath(String name) {
         LOGGER.exiting(files.toString());
         return files;
     }
+
+    /**
+     * Installs MicrosoftWebDriver.msi file as administrator in SELION_HOME_DIR and also deletes unwanted file and
+     * directory created by Msiexec.
+     *
+     * @param msiFile
+     * @return
+     */
+
+    static String extractMsi(String msiFile) {
+        Process process = null;
+
+        try {
+            process = Runtime.getRuntime().exec(
+                    new String[] { "msiexec", "/a", msiFile, "/qn", "TARGETDIR=" + SeLionConstants.SELION_HOME_DIR });
+            process.waitFor();
+        } catch (IOException e) {
+            LOGGER.log(Level.SEVERE, "Could not find file " + msiFile, e);
+        } catch (InterruptedException e) {
+            LOGGER.log(Level.SEVERE, "Exception waiting for msiexec to be finished", e);
+        } finally {
+            process.destroy();
+        }
+        moveExeFileToSeLionHomeAndDeleteExtras();
+        /*
+         * Collection<File> listFiles = FileUtils.listFiles(new File(SeLionConstants.SELION_HOME_DIR), new String[] {
+         * "exe" }, false);
+         */
+        String exeFilePath = FileUtils.getFile(SeLionConstants.SELION_HOME_DIR + "MicrosoftWebDriver.exe")
+                .getAbsolutePath();
+        LOGGER.exiting(exeFilePath);
+        return exeFilePath;
+
+    }
+
+    static void moveExeFileToSeLionHomeAndDeleteExtras() {
+
+        File source = new File(SeLionConstants.SELION_HOME_DIR + File.separator + "Microsoft Web Driver"
+                + File.separator + "MicrosoftWebDriver.exe");
+        File dest = new File(SeLionConstants.SELION_HOME_DIR);
+
+        try {
+            FileUtils.copyFileToDirectory(source, dest, true);
+        } catch (IOException e) {
+            LOGGER.log(Level.SEVERE,
+                    "Error while moving MicrosoftWebDriver.exe file  from '" + source.getAbsolutePath() + "' to  '"
+                            + dest.getAbsolutePath() + "'", e);
+        }
+
+        // Deleting "Microsoft Web Driver" directory after moving file from this folder to SELION_HOME_DIR
+        try {
+            FileUtils.deleteDirectory(new File(SeLionConstants.SELION_HOME_DIR + File.separator
```

Comment:
FileUtils.moveFile()  deletes only file not the folder and  "Microsoft Web Driver" folder has to get deleted, and to move files FileUtils.copyFileToDirectory(source, dest, true) is serving purpose.

Please let me know if i am wrong or if moveFile has advantages over copyFileToDirectory.